cmd/clusterctl/client/cluster/objectgraph.go

```diff
@@ -92,6 +104,7 @@ func (n *node) isSoftOwnedBy(other *node) bool {
 type objectGraph struct {
 	proxy     Proxy
 	uidToNode map[types.UID]*node
+	crds      map[string]*crdInfo
```

Comment:
I found myself needing to pass around the full crdInfo map just to look objects up and see if they were supposed to be force moved. Adding this map to the objectGraph seemed easier. Note that it changed the return values of getDiscoveryTypes, so that's updated in a bunch of places.